complaint_search/es_interface.py

```diff
@@ -15,20 +15,130 @@
 _COMPLAINT_ES_INDEX = os.environ.get('COMPLAINT_ES_INDEX', 'complaint-index')
 _COMPLAINT_DOC_TYPE = os.environ.get('COMPLAINT_DOC_TYPE', 'complaint-doctype')
 
+_OPTIONAL_FILTERS = ("product", "issue", "company", "state", "zip_code", "timely", 
+    "company_response", "company_public_response", 
+    "consumer_consent_provided", "submitted_via", "tag")
+
+_OPTIONAL_FILTERS_PARAM_TO_ES_MAP = {
+    "product": "product.raw",
+    "sub_product": "sub_product.raw",
+    "issue": "issue.raw",
+    "sub_issue": "sub_issue.raw"
+}
+_OPTIONAL_FILTERS_CHILD_MAP = {
+    "product": "sub_product", 
+    "issue": "sub_issue"
+}
+
+_OPTIONAL_FILTERS_STRING_TO_BOOL = ("consumer_disputed", "has_narratives")
+
 def get_es():
     global _ES_INSTANCE
     if _ES_INSTANCE is None:
         _ES_INSTANCE = Elasticsearch([_ES_URL], http_auth=(_ES_USER, _ES_PASSWORD), 
             timeout=100)
     return _ES_INSTANCE
 
-def _create_and_append_bool_should_clauses(es_field_name, value_list, 
-    filter_list, with_subitems=False, es_subitem_field_name=None):
+
+
+def _create_aggregation(**kwargs):
+
+    Field = namedtuple('Field', 'name title size has_subfield')
+    fields = [
+        Field('has_narratives', 'Only show complaints with narratives?', 10, False),
+        Field('company', 'Matched company name', 10000, False),
+        Field('product', 'Product / Subproduct', 10000, True),
+        Field('issue', 'Issue / Subissue', 10000, True),
+        Field('state', 'State', 50, False),
+        Field('zip_code', 'Zip code', 1000, False),
```

Comment:
@JeffreyMFarley This is the size of the results being returned, not the # of characters in the zip code field. I will also take a look at what makes sense for each field as well.